apps/system/js/advanced_telemetry.js

```diff
@@ -324,6 +362,84 @@
     }
   };
 
+  // Check if there are existing metrics that need to be merged, if so,
+  // merge them and then transmit.
+  AT.prototype.mergeSend = function mergeSend(payloadNew, send) {
```

Comment:
Perhaps the code would be cleaner if this function simply returned a boolean whether data was merged or not.